q/verbose/verbose.q

```diff
@@ -0,0 +1,471 @@
+//
+.finos.verbose.fixOrdersGEN:{[o; t]
+    .finos.verbose.fixColumnOrderGEN[o] .finos.verbose.fixRowOrderGEN[o;t]
+ };
+
+///
+// does asof aggregation of a value keyed by a certain column
+// Let us assume that we have a table that contains columns time, sym
+// and a field that would would like to sum for each sym.
+// The problem is that the value is available for different time stamps.
+// This function bring the input to a common key base (generally common date and time field)
+// and applies the aggregate function (e.g. addition).
+// This function can be used for example to calculate the midprice of and index (weighted set of instruments).
+//
+.finos.verbose.aggregateFieldValuesAsOf: {[t; k; p; v; colname; binop]
+  piv: fills .ms.pivot.pvtfw[t; k; p; v; (); ()];
+  c: (cols piv) except cols t;
+
+  // if single key value just rename column, otherwise apply binary operator
+  // I am sure the exists a more elegant solution.
+  colCalculus: $[1 = count c;
+    first c;
+    enlist (first c) {(x; z ;y)}[binop]/ 1 _ c];
+
+  :?[piv; (); 0b; (k, colname)!k, colCalculus];
+  }
+
+///
+// "n" seconds window before t; Implies [t-n Seconds; t] time window.
+// @param n Number of seconds
+// @param t Time field - it will be used in wj
+// @return pairs of time (indicating the interval)
+.finos.verbose.nSecWinBefT:{[n;t] (t - n*1000;t)};
+
+///
+// "n" seconds window after t; Implies [t t+n seconds] time window.
+.finos.verbose.nSecWinAftT:{[n;t] (t; t + n*1000)};
+///
+// "n" minutes window before t; Implies [t-n minutes; t] time window
+.finos.verbose.nMinWinBefT:{[g;n;t] g[60*n;t]}[.finos.verbose.nSecWinBefT];
+///
+// "n" minutes window after t; Implies [t t+n minutes] time window.
+.finos.verbose.nMinWinAftT:{[g;n;t] g[60*n;t]}[.finos.verbose.nSecWinAftT];
+///
+// "n" hours window before t; Implies [t-n hours; t] time window
+.finos.verbose.nHrWinBefT:{[g;n;t] g[3600*n;t]}[.finos.verbose.nSecWinBefT];
+///
+// "n" hours window after t; Implies [t t+n hours] time window.
+.finos.verbose.nHrWinAftT:{[g;n;t] g[3600*n;t]}[.finos.verbose.nSecWinAftT];
+
+// filter out commas in column values, e.g. for QlikView
+.finos.verbose.dropCommas:{[t]
+  :flip {[col]$[(10h=type first col) and 0h=type col;ssr[;",";""]'[col];11h=type col;`$ssr[;",";""]'[string col];col]} each flip t
+  };
+
+.finos.verbose.asyncQuery:{[query;callbackFunName;callbackId]
+    res:try[{(`success;.z.pg x)}; enlist query;{(`error;x)}];
+    neg[.z.w](callbackFunName;callbackId),res;
+    neg[.z.w][::];
+    };
+
+.finos.verbose.localTableMeta:{[table]
+    if[not .Q.qt table; '".finos.verbose.localTableMeta: expects table"];
+    meta table};
+
+//security-safe version of try
+.finos.verbose.try:{[fun;params;errorHandler]
+    if[100h<>type fun; '"fun must be a lambda"];
+    if[100h<>type errorHandler; '"errorHandler must be a lambda"];
+    if[not type[params] within 0 20h; '"params must be a list"];
+    try[fun;params;errorHandler]};
```

Comment:
I think there is no global `try` function yet.

Reply:
Will put in .finos.errorTrapDot and use that instead.